src/garbage_collector.rs

```diff
@@ -129,24 +130,46 @@ struct GarbageCollectorOrchestrator {
 }
 
 impl GarbageCollectorOrchestrator {
-    /// Collect garbage from the manifest store. This will delete any manifests
-    /// that are older than the minimum age specified in the options.
-    async fn collect_garbage_manifests(&self) -> Result<(), SlateDBError> {
-        let utc_now = Utc::now();
+    fn manifest_min_age(&self) -> chrono::Duration {
         let min_age = self
             .options
             .manifest_options
             .map_or(DEFAULT_MIN_AGE, |opts| opts.min_age);
+        chrono::Duration::from_std(min_age).expect("invalid duration")
+    }
+
+    /// Collect garbage from the manifest store. This will delete any manifests
+    /// that are older than the minimum age specified in the options.
+    async fn collect_garbage_manifests(&self) -> Result<(), SlateDBError> {
+        self.remove_expired_checkpoints().await?;
```

Comment:
A problem similar to the one @rodesai mentioned in #352 could occur with manifest as well, if the writer or compactor has stalled for a while.

Consider the example of three manifests [1, 2, 3]. 3 refers to 1. GC deletes 2.
A compactor or a wal disabled writer is currently at 1, it writes 2, which succeeds since 2 is deleted.
One solution could be to list, and get the recent manifest id, to confirm if there is no newer manifest.

@jayaprabhakar , these are cases we should consider whenever fizzbee protocol is updated with GC.

Reply:
Yeah, interesting case. The minimum age adds some protection in the common case . Perhaps the garbage collector could leave behind a tombstone which preserves epoch information, but drops all checkpoints in order to allow them to be cleaned up. Something like that?

Reply:
> Yeah, interesting case. I guess the minimum age adds some protection in the common case

It definitely should. The poll intervals are on the order of seconds, and the min_age is a day. I do like the idea of retaining tombstones instead of just deleting the objects (in both the WAL and the manifest case). The challenge then becomes listing the manifests when GC'ing. We'd need to maintain some watermark beyond which we know we've deleted everything to help prune the search. Another approach here would be to change how we write the manifest by adding a layer of indirection to the manifest object - so there's something like a `manifest.current` file that always points to the current manifest and needs to be updated to complete a manifest write.